db/controllers/controller.js

```diff
@@ -0,0 +1,25 @@
+const db = require("../connection");
+
+const fetchTopicsFromDatabase = async () => {
+  try {
+    const queryResult = await db.query("SELECT * FROM topics;");
+    return queryResult.rows;
+  } catch (error) {
+    throw new Error("Error getting topics: " + error.message);
+  }
+};
+
+const getTopics = async (req, res, next) => {
+  try {
+    const topics = await fetchTopicsFromDatabase();
+    if (topics.length === 0) {
+      res.status(404).send({ error: "No topics found" });
+    } else {
+      res.json(topics);
+    }
+  } catch (error) {
```

Comment:
I would invoke next here with the error and handle in app.js with our error handling middleware chain